internal/client/tasks.go

```diff
@@ -2,41 +2,60 @@ package client
 
 import (
 	"github.com/sirupsen/logrus"
-	"golang.org/x/sync/semaphore"
-
-	// "github.com/sirupsen/logrus"
-	fleetdbapi "github.com/metal-toolbox/fleetdb/pkg/api/v1"
 )
 
-func (c *Client) CreateConditionInventoryForAllServers(pageSize, inFlightPages int) error {
-	// Start thread to start collecting servers
-	serverCh, concLimiter, err := c.GatherServersNonBlocking(pageSize, inFlightPages)
+func (c *Client) CreateConditionInventoryForAllServers(pageSize int) error {
```

Comment:
Since most of fleet-schedulers work is going to be - iterate over servers in fleetdb, filter on a given criteria - we'd like this pattern to be reusable and have tests.

Could this code be structured in an iterator like pattern and we have tests added - as in the iterator part is in a separate method ?
theres some ideas here https://pkg.go.dev/google.golang.org/api/iterator

This can be done as a separate effort/PR.

Reply:
Ticket Created